flake.nix

```diff
@@ -0,0 +1,100 @@
+{
+  description = "LSP-AI: Open-source language server for AI-powered functionality";
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable";
+    flake-utils.url = "github:numtide/flake-utils";
+    rust-overlay.url = "github:oxalica/rust-overlay";
+  };
+
+  outputs =
+    {
+      self,
+      nixpkgs,
+      flake-utils,
+      rust-overlay,
+    }:
+    flake-utils.lib.eachDefaultSystem (
+      system:
+      let
+        overlays = [ (import rust-overlay) ];
+        pkgs = import nixpkgs {
+          inherit system overlays;
+          config.allowUnfree = true;
+        };
+        rustVersion = pkgs.rust-bin.stable.latest.default;
+
+        commonBuildInputs = with pkgs; [
+          openssl
+          zlib
+        ];
+
+        commonNativeBuildInputs = with pkgs; [
+          pkg-config
+          cmake
+          rustVersion
+          perl
+        ];
+
+      in
+      rec {
+        packages = rec {
+          default = pkgs.rustPlatform.buildRustPackage {
```

Comment:
This is the `main` package. Essentially when running `nix build` locally, the `default` entry is the starting point. The devPackage and devShells are more of conveniences for being dropped into a shell with all needed dependencies, but still track a cached build at `/tmp/.cargo` so during the dev cycle, one can run `cargo` commands as they please with the necessary caches.